models/WebsiteSetting.php

```diff
@@ -274,6 +278,29 @@ public function delete(): void
             unset(self::$nameIdMappingCache[$nameCacheKey]);
         }
 
+        $event = new WebsiteSettingEvent($this);
+
+        $this->dispatchEvent($event, WebsiteSettingEvents::PRE_DELETE);
+
         $this->getDao()->delete();
+
+        $this->dispatchEvent($event, WebsiteSettingEvents::POST_DELETE);
+    }
+
+    public function save(): void
+    {
+        $event = new WebsiteSettingEvent($this);
+
+        $websiteSetting = clone $this;
+
+        $this->dispatchEvent($event, $websiteSetting->id === null ?
+            WebsiteSettingEvents::PRE_ADD : WebsiteSettingEvents::PRE_UPDATE
+        );
+
+        $this->getDao()->save();
+
+        $this->dispatchEvent($event, $websiteSetting->id === null ?
+            WebsiteSettingEvents::POST_ADD : WebsiteSettingEvents::POST_UPDATE
+        );
```

Comment:
I don't think we need a heavy clone just for checking one property:

```suggestion
        $isAdd = $this->id === null;

        $this->dispatchEvent($event, $isAdd ? WebsiteSettingEvents::PRE_ADD : WebsiteSettingEvents::PRE_UPDATE
        );

        $this->getDao()->save();

        $this->dispatchEvent($event, $isAdd ? WebsiteSettingEvents::POST_ADD : WebsiteSettingEvents::POST_UPDATE
        );
```